src/storage/mapping/BaseFileIdentifierMapper.ts

```diff
@@ -33,12 +34,23 @@ export class BaseFileIdentifierMapper implements FileIdentifierMapper {
    * Determines the content type if none was provided.
    * For containers the content-type input is ignored.
    * @param identifier - The input identifier.
+   * @param isMetadata - If we need the data or metadata file path.
    * @param contentType - The content-type provided with the request.
    *
    * @returns A ResourceLink with all the necessary metadata.
    */
-  public async mapUrlToFilePath(identifier: ResourceIdentifier, contentType?: string): Promise<ResourceLink> {
-    const path = this.getRelativePath(identifier);
+  public async mapUrlToFilePath(identifier: ResourceIdentifier, isMetadata: boolean, contentType?: string):
```

Comment:
Duh, of course I should have done that. I first had it as an optional parameter for the same reason but that looked wrong 😂

Reply:
Ah no nevermind, you can't do that in an interface so it would require the parameter to be optional in the interface which feels a bit more iffy (since that allows both undefined and false and doesn't clearly show that they are the same in the interface typings).